### ISSUE TITLE
Custom dataset

### ISSUE THREAD

**Issue by damonftl** [buildingnet/buildingnet_dataset#8]
Hi, I'd like to try and make my own dataset of different buildings with parts other than the ones included with BuildingNet. Generating the 3D models (and point clouds) themselves shouldn't be a problem - it's everything else I'm not sure about.

For the minkowski pretrained features, should I simply download their code (https://github.com/NVIDIA/MinkowskiEngine) and run the models through it?

The required json files for each model used to train BuildingNet are fairly extensive (adjacency, containment, support, and similarity) and they don't seem especially trivial to recreate. The code for their creation doesn't appear to be in this repository, and I'm guessing it is part of the labeling application shown in the paper. Can the code for the labeling application be downloaded? I'm hoping in the end to output the label data automatically together with the 3D models when they are generated (that is, a synthetic training data generation pipeline) and being able to see some of that code would be a huge help.

**Comment by damonftl** [comment]
Hello mzillag,
I did not end up making the pipeline back at the time I originally posted this. However, I did contact the authors of the paper and they have since uploaded more of the code required here to GitHub. The feature scripts and annotation tool are here now, which should make it possible to use your own custom data. Unfortunately the project I was going to use this for ran out of time before I could really try it.

However, I have a new project starting very soon that will use this, so I'll be getting back to it and making another attempt at using custom data. If I run into anything I'll be sure to post it here.

**Comment by mzillag** [comment]
Thanks for the answer. So you didn’t know how to test the model on point cloud (which has not labeled) not from their dataset?

**Comment by damonftl** [comment]
That's right, exactly. I was able to train and test fine with their dataset, but not do either with my own. Hopefully soon though!
